File: u1/Sample_Code/DemoUDPReceiver.py
import socket

def main():
	sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM) #create datagram socket
	sock.bind(('127.0.0.1', 12345)) #bind to port 12345
	print ('Waiting for connection in')
	
	while True:
		(line, address) = sock.recvfrom(1024) #Receiver receive data and address from socket
		print ('{a} Message from sender [{h}] {a}'.format(a='*'*10, h=address[0]))
		print (line.decode('utf-8').strip()) #process it

if __name__ == "__main__":
	main()

File: u1/UDPSocketProgramming/src/udpdemo01/DemoUDPReceiver.java
package udpdemo01;

import java.io.*;
import java.net.*;
import java.util.*;

/**
 *
 * @author Cyrus Cheng
 */
public class DemoUDPReceiver{
	
    public static void main(String args[]){
        try{
            DatagramSocket _socket = new DatagramSocket(12345); //create the datagram socket

            byte[] _buffer = new byte[255];//prepare buffer to feed into datagram packet
            DatagramPacket _packet = new DatagramPacket(_buffer, _buffer.length);//create the datagram packet
            System.out.println("Waiting for connection in");//wait for connection in

            while(true){
                Arrays.fill(_buffer, (byte)0); //reset buffer for each receive
                _socket.receive(_packet);//get the data from datagram packet through socket
                System.out.printf("Connection with %s created%n", _packet.getAddress());
                System.out.println("********** Message from sender **********");
                System.out.println(new String(_buffer));//process it
            }
        }catch(IOException ioe){
            ioe.printStackTrace();
        }catch(Exception e){
            e.printStackTrace();
        }
    }
	
}
